### DISCUSSION TITLE
👋 Help test the next major version of Gatsby!

### DISCUSSION THREAD

**Discussion by pragmaticpat** [gatsbyjs/gatsby#29566] [Umbrella Discussions]
## Greetings, Gatsby community members!

We're working on the next major version of Gatsby and would **love** your help as we work out any rough spots! So, grab it while it's 🔥, and help us find those edges we may have missed!

1. Set your `gatsby` dependency to `next` in your project's `package.json` file:

    ```json
    {
      "dependencies": {
        "gatsby": "next",
      },
    }
    ```
1. Update any `gatsby-*` plugin that we publish / is in [our monorepo](https://github.com/gatsbyjs/gatsby/tree/master/packages) to `next`, too. For example, if you use `gatsby-transformer-sharp`:

    ```json
    {
      "dependencies": {
        "gatsby-transformer-sharp": "next",
      },
    }
    ```
1. Remove the `node_modules` and `.cache` folders within the project to ensure all dependencies are updated appropriately as well.
1. Upgrade Gatsby to the next version by running `npm i` in the project directory for your Gatsby website
1. Confirm you've upgraded by running `gatsby -v` and ensuring that you see `Gatsby version: 3.0.0-next.x` ("x" will increment as we push patches) in the output.
1. Start your dev server in typical fashion via `gatsby develop --verbose` (use the `--verbose` option to ensure you receive verbose output from the development server)
1. Make source and content edits & confirm that you're seeing the corresponding updates in the dev site that's running
1. Stop your dev server
1. **Build** your site in typical fashion via `gatsby build --verbose` (again leveraging the `--verbose` option to ensure verbose output, should there be any issues)
1. Make more source and content edits
1. Build again, and ensure that you see the edits in the resulting site
1. Repeat steps 9 & 10 a few times

## Where do I find more detailed information of the key updates to help me test?
You can find more information in our in-process [migration guide](https://github.com/gatsbyjs/gatsby/blob/docs/v3-migration-guide/docs/docs/reference/release-notes/migrating-from-v2-to-v3.md)

## ☠️ What if something went wrong?

Report issues right here in this discussion so everyone can benefit from the findings.

## Thanks, everyone! 🙏

**Comment by nibtime** [comment]
is there some document that lists all the brand new features of Gatsby v3, i.e. all the things that aren't covered by minor and patch releases?

That would help me a lot in testing it out. I could then look through my existing Gatsby projects and try to find potential use cases for those new features.

**Reply by pragmaticpat** [reply]
Thanks for reaching out and asking! 

You can find our in-process migration guide [here](https://github.com/gatsbyjs/gatsby/blob/docs/v3-migration-guide/docs/docs/reference/release-notes/migrating-from-v2-to-v3.md), which illustrates areas of breaking change, etc.

Being a work in process, there are a couple of dead link / placeholder links to other docs, which are _also_ in process.

The tldr; of key changed areas include:
- GraphQL
- Webpack 5 update
- Node update
- Changes to the local rebuild / warm build process

As noted in the original post, above, there are also build performance enhancements, which steps 8 - 12 in the original post are meant to help you test.

I'm updating the original post with a link to the in-process migration guide so there's less of a need to comb through the discussion to find that information.

Hope this helps, and definitely reach back out here if you have any more questions. Thanks for testing! 💜

**Reply by kamranayub** [reply]
The build time will be a good test. For https://github.com/excaliburjs/excaliburjs.github.io our build tends to take awhile (probably my fault) as we run some custom plugins for transforming Typedoc. Right now `run queries` takes about 75s so I may be able to update the plugins/site with V3 to test improvements.

**Comment by pragmaticpat** [comment]
Hey folks! Now that [Gatsby V3](https://www.gatsbyjs.com/blog/gatsby-v3) is live, we're locking this discussion. You can still provide feedback on Gatsby V3 through our standard process within the Github project.

* If you encounter an issue with a community plugin when using Gatsby V3, please add to the [Gatsby V3 Plugins discussion](https://github.com/gatsbyjs/gatsby/discussions/categories/gatsby-v3-plugin-issues) to help with the triage and coordination with the plugin authors
* If you encounter other issues upgrading / using Gatsby V3, please file a bug through the [standard bug submission process](https://github.com/gatsbyjs/gatsby/issues/new?assignees=&labels=type%3A+bug&template=bug_report.md)

Thanks everyone!
